spec/unit/file_system_spec.rb

```diff
@@ -999,6 +999,15 @@ def increment_counter_in_multiple_processes(file, num_procs, options)
             Puppet::FileSystem.replace_file(dest, 0755) { |_| }
           }.to raise_error(ArgumentError, /Only modes 0644, 0640, 0660, and 0440 are allowed/)
         end
+
+        it 'falls back to fully qualified user name when sid retrieval fails' do
+          current_user_sid = Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+          allow(Puppet::Util::Windows::SID).to receive(:name_to_sid).with(Puppet::Util::Windows::ADSI::User.current_user_name).and_return(nil, current_user_sid)
+          allow(Puppet::Util::Windows::SID).to receive(:name_to_sid).with(Puppet::Util::Windows::ADSI::User.current_user_name_with_format(Puppet::Util::Windows::ADSI::User::NameSamCompatible)).and_call_original
```

Comment:
I think you don't need this stub, as `call_original` will just pass through, and the method will be called.

Reply:
I actually need this since `name_to_sid` is called multiple times and will end up with:
```
 Puppet::Util::Windows::SID received :name_to_sid with unexpected arguments
         expected: ("Administrator")
              got: ("MYPC\\Administrator")
        Please stub a default value first if message might be received with other args as well.
```
In my test I want only the first call to return `nil` and force the fallback to take over. After that, I want the checks to use original calls to prove that we will get the same SID in the end, even with the fallback (We can't cover the whole complex case in our tests so this should do).

Reply:
I read it wrong, makes sense. 👍